packages/@lwc/integration-karma/test/component/LightningElement.attachInternals/api/index.spec.js

```diff
@@ -13,74 +13,91 @@ const testConnectedCallbackError = (elm, msg) => {
     expect(error.message).toBe(msg);
 };
 
-const createTestElement = (name, def) => {
-    const elm = createElement(name, { is: def });
-    document.body.appendChild(elm);
-    return elm;
+const createElementsThroughLwcAndCustomElementConstructor = (tagName, ctor) => [
+    createElement(`ai-${tagName}`, { is: ctor }),
+    createCustomElementUsingCec(`cec-ai-${tagName}`, ctor.CustomElementConstructor),
+];
+
+const createCustomElementUsingCec = (tagName, ctor) => {
+    if (!customElements.get(tagName)) {
+        customElements.define(tagName, ctor);
+    }
+    return document.createElement(tagName);
 };
 
-const attachInternalsSanityTest = (cmp) => {
-    let elm;
-    beforeEach(() => {
-        elm = createTestElement('ai-component', cmp);
-    });
+const attachInternalsSanityTest = (tagName, ctor) => {
+    createElementsThroughLwcAndCustomElementConstructor(
+        `${tagName}-element-internal-enabled`,
+        ctor
+    ).forEach((elm) => {
+        beforeAll(() => {
+            document.body.appendChild(elm);
+        });
 
-    afterEach(() => {
-        document.body.removeChild(elm);
-    });
+        afterAll(() => {
+            document.body.removeChild(elm);
+        });
 
-    it('should be able to create ElementInternals object', () => {
-        expect(elm.hasElementInternalsBeenSet()).toBeTruthy();
-    });
+        it('should be able to create ElementInternals object', () => {
+            expect(elm.hasElementInternalsBeenSet()).toBeTruthy();
+        });
 
-    it('should throw an error when called twice on the same element', () => {
-        // The error type is different between browsers
-        expect(() => elm.callAttachInternals()).toThrowError();
+        it('should throw an error when called twice on the same element', () => {
+            // The error type is different between browsers
+            expect(() => elm.callAttachInternals()).toThrow();
```

Comment:
I'm comfortable with using a regex here to capture the different browser error messages. This is what we've historically done for these kinds of cases.